testing/kuttl/e2e/pgbackrest-restore/14--lose-data.yaml

```diff
@@ -0,0 +1,55 @@
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestStep
+commands:
+  - script: |
+      PRIMARY=$(
+        kubectl get pod --namespace "${NAMESPACE}" \
+          --output name --selector '
+            postgres-operator.crunchydata.com/cluster=original,
+            postgres-operator.crunchydata.com/role=master'
+      )
+      OBJECTIVE=$(
+        kubectl exec --namespace "${NAMESPACE}" "${PRIMARY}" \
+          -- psql -qAt --command 'SELECT clock_timestamp()'
+      )
+
+      # Store the recovery objective for later steps.
+      kubectl annotate --namespace "${NAMESPACE}" postgrescluster/original \
+        "testing/objective=${OBJECTIVE}"
+
+      # A reason to restore. Wait for the change to be sent to the WAL archive.
+      kubectl exec --namespace "${NAMESPACE}" "${PRIMARY}" \
+        -- psql -qb original --set ON_ERROR_STOP=1 \
+          --command 'DROP TABLE important' \
+          --command "SELECT pg_stat_reset_shared('archiver')" \
+          --command '
+            DO $$
+            BEGIN
+              LOOP
+                PERFORM pg_switch_wal() FROM pg_stat_archiver WHERE archived_count = 0;
+                EXIT WHEN NOT FOUND;
+                PERFORM pg_sleep(1); -- seconds
+                ROLLBACK; -- fresh stats
+              END LOOP;
+            END $$'
+
+  # The replica should also need to be restored.
+  - script: |
+      REPLICA=$(
+        kubectl get pod --namespace "${NAMESPACE}" \
+          --output name --selector '
+            postgres-operator.crunchydata.com/cluster=original,
+            postgres-operator.crunchydata.com/role=replica'
+      )
+
+      kubectl exec --stdin --namespace "${NAMESPACE}" "${REPLICA}" \
+        -- psql -qb original --set ON_ERROR_STOP=1 \
+          --file=- <<'SQL'
+        DO $$
+        BEGIN
+          ASSERT to_regclass('important') IS NULL, 'expected no table';
+          PERFORM * FROM information_schema.tables WHERE table_name = 'important';
+          ASSERT NOT FOUND, 'expected no table';
```

Comment:
For my own education, is there a case where line 51 would pass, but 53 would fail?

Reply:
Not that I know of. I did both because they both seem kinda obscure. `to_regclass` is the most direct but the least obvious to me.

I chose "dropped the wrong table" as a typical data-loss scenario but writing a positive assertion that it happened is awkward. I considered also changing from `DROP` to `DELETE` so that these assertions could be "no rows in table". I think the full assertion here would become: (no regclass)

```plpgsql
PERFORM * FROM important;
ASSERT NOT FOUND, 'expected no data';
```

Would that be preferable?

Reply:
Speaking of "obscure," I definitely had to go look up `to_regclass`, but I think this test is good.

(I'm trying to remember whether I have more often erased rows I didn't mean to or dropped whole tables by accident, and it's probably 50/50 in PG. (In Mongo, it's much more often to hear someone say, "oh no, I erased how many documents?") So I think dropping the table is a good test scenario.)